okapi/services/fieldnotes/upload/WebService.php

```diff
@@ -0,0 +1,175 @@
+<?php
+
+namespace okapi\services\fieldnotes\upload;
+
+use okapi\core\Exception\InvalidParam;
+use okapi\core\Exception\ParamMissing;
+use okapi\core\Db;
+use okapi\core\Okapi;
+use okapi\core\OkapiServiceRunner;
+use okapi\core\Request\OkapiInternalRequest;
+use okapi\core\Request\OkapiRequest;
+use okapi\services\logs\LogsCommon;
+use okapi\Settings;
+
+class WebService
+{
+    public static function options()
+    {
+        return array(
+            'min_auth_level' => 3
+        );
+    }
+
+    public static function call(OkapiRequest $request)
+    {
+        $result = array(
+            'success' => false   // if the installation doesn't support it
+        );
+
+        if (Settings::get('OC_BRANCH') == 'oc.de')
+        {
+
+            $field_notes = $request->get_parameter('field_notes');
+            if (!$field_notes) throw new ParamMissing('field_notes');
+
+            $notes = self::parse_notes($field_notes);
+            if ($notes['success'] === false) throw new InvalidParam('field_notes', "Input data not recognized.");
```

Comment:
I understand the point. Obviously you have been confronted with this type of complaints.

Do you have a suggestion how this could be made better?  Since we're throging an exception, I can handle it deeper down and not even let it bubble up to this point. .... Let me make an attempt ....

Reply:
IMO best practice is to throw and catch exceptions.